File: lab3/lib/viewmodel/authorization_viewmodel.dart
import 'package:firebase_auth/firebase_auth.dart';

class AuthorizationViewModel {
  final FirebaseAuth _auth = FirebaseAuth.instance;

  bool isShowFilmsListView = false;
  bool isShowAlert = false;

  Future<void> authorization(String email, String password) async {
    if (password.length < 8 || email.isEmpty) {
      return;
    }

    await _auth.signInWithEmailAndPassword(
      email: email,
      password: password,
    );
    isShowFilmsListView = true;
  }
}

File: lab1/CoolFilms/CoolFilms/ViewModel/AuthorizationViewModel.swift
import Foundation
import FirebaseAuth

class AuthorizationViewModel: ObservableObject {
    @Published var email = ""
    @Published var password = ""
    @Published var alert = ""
    @Published var isShowFilmsListView = false
    @Published var isShowAlert = false
    
    private let auth = Auth.auth()
    
    func hidePopup(){
        isShowAlert = false
    }
    
    func showAlert(message: String) {
        alert = message
        isShowAlert = true
    }
    
    func authorization() {
        guard password.count >= 8, !email.isEmpty else {
            showAlert(message: "Invalid data in input fields")
            return
        }
        
        auth.signIn(withEmail: email, password: password) { (authResult, error) in
            if let error = error {
                self.showAlert(message: error.localizedDescription)
                return
            }
            
            self.isShowFilmsListView = true
        }
    }
}
